dvc/stage.py

```diff
@@ -486,19 +486,24 @@ def _check_dvc_filename(fname):
             )
 
     @staticmethod
-    def _check_file_exists(fname):
-        if not os.path.exists(fname):
+    def _check_file_exists(repo, fname, branch):
+        if not repo.scm.exists(fname, branch):
             raise StageFileDoesNotExistError(fname)
 
     @staticmethod
-    def load(repo, fname):
-        Stage._check_file_exists(fname)
+    def load(repo, fname, branch=None):
+        # raise proper exceptions
+        Stage._check_file_exists(repo, fname, branch)
         Stage._check_dvc_filename(fname)
-
-        if not Stage.is_stage_file(fname):
-            raise StageFileIsNotDvcFileError(fname)
-
-        d = load_stage_file(fname)
+        if not branch:
+            # for other errors (like file is a directory)
+            if not Stage.is_stage_file(fname):
+                raise StageFileIsNotDvcFileError(fname)
+            d = load_stage_file(fname)
+        else:
+            d = load_stage_file_fobj(
+                repo.scm.open(fname, branch=branch), fname
```

Comment:
if we use `branch` as a way to distinguish direct access vs API, I think we should use it more consistently ... for example, why don't make  `repo.scm.open(fname, branch=branch)` open the file the regular way if branch is None. You are already utilizing this approach for some other methods as far as I understand. Basically, it's better to keep stuff like `if not branch` in a single place.